pmd-java/pom.xml

```diff
@@ -120,6 +120,12 @@
             <artifactId>commons-lang3</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
```

Comment:
I don't think they're "strictly illegal", unless there's something special about  `javax` packages that I don't know?

Ultimately the goal of using this is mostly documentation. What makes annotations superior to just a comment is
* They improve the kotlin API
* Some IDEs (Eclipse, IntelliJ) support some form of nullness checking. The specific annotation set they use to perform their analysis is configurable, so whichever artifact we chose is ok.  
* IMHO it's much easier to find out at a glance what to expect from a method if eg the params are annotated with `@Nullable` or `@Nonnull`, than if one has to read through the javadoc to find out. It's also easier to write than a sentence in English.

Btw I think in fact, this artifact is the de facto "standard", at least it's more standard than eg `org.jetbrains.annotations`. See https://stackoverflow.com/questions/4963300/which-notnull-java-annotation-should-i-use?rq=1

Reply:
Both `java` and `javax` packages are reserved for Java itself and it's extensions respectively.

Actually, Spotbugs (the Findbugs fork and successor) removed them early on because of this: https://github.com/spotbugs/spotbugs/pull/180

Maybe JSR-308 (checker framework) is a better option, as that one is approved.

Reply:
Ah ok that makes sense. From what I understand jsr-308 is about type annotations. Does it encompass the checker framework annotations? Or does using it just mean we have to include an artifact from the checker framework project?

Reply:
No, the checkerframework seems to have no separate artifact for annotations…

Reply:
Checkerframework has `org.checkerframework:checker-qual` artifact for annotations.

Reply:
@olegshtch True, thanks! 

That artifact also contains all the other annotations of the checker framework. OTOH it's the only annotation set that provides a [default qualifier](https://checkerframework.org/manual/#null-defaults) annotation which doesn't rely on JSR 305's `TypeQualifierDefault`. For instance,  `org.jetbrains:annotations` [recommends depending on jsr305](https://www.jetbrains.com/help/idea/parametersarenonnullbydefault-annotation.html), and [`spotbugs-annotations`](https://search.maven.org/artifact/com.github.spotbugs/spotbugs-annotations/4.0.0-beta1/jar) has it as a POM dependency... 

So checker-qual is probably the best bet.